### PR TITLE
BUG: compile np.array(list_of_arrays)

### PR DESCRIPTION
**Pull request by ev-br**
Add a shortcut for a sequence of arrays only. This remove a graph break on a common pattern of
`np.array([np.cos(theta), np.sin(theta)])` and its ilk.

This PR is a simpified alternative to https://github.com/pytorch/pytorch/pull/112521 --- it still breaks on mixing arrays and scalars or array_likes (e.g.  `np.array([[1, 2], np.array[3, 4]])`) and instead adds a simple shortcut.

cc @mruberry @rgommers @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: torch/_numpy/_ndarray.py [hunk 1]
```diff
@@ -505,8 +505,13 @@ def array(obj, dtype=None, *, copy=True, order="K", subok=False, ndmin=0, like=N
     ):
         return obj
 
-    # lists of ndarrays: [1, [2, 3], ndarray(4)] convert to lists of lists
     if isinstance(obj, (list, tuple)):
+        # list of arrays *under torch.Dynamo*
+        if obj and all(isinstance(x, torch.Tensor) for x in obj):
+            return ndarray(torch.stack(obj))
```
Comment (lezcano) [inline torch/_numpy/_ndarray.py:511]: this should be a recursive call into `array` with all the args and kwargs it takes.

Comment (ev-br) [reply torch/_numpy/_ndarray.py]: Good catch, done.

**Review by lezcano — APPROVED** [review]
Great, thanks!

**Pull request closed without merging**